From 04236a7a0c6bd50337da45b364c5bd1d86be03ef Mon Sep 17 00:00:00 2001
In-Reply-To: <1419964529-25087-1-git-send-email-psusi@ubuntu.com>
References: <1419964529-25087-1-git-send-email-psusi@ubuntu.com>
From: "Brian C. Lane" <bcl@redhat.com>
Date: Wed, 25 Feb 2015 14:37:40 -0800
Subject: [PATCH] libparted: Use pri before freeing it.

This could have used pri after freeing it. Move it up into the if
clause.
---
 libparted/labels/gpt.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/libparted/labels/gpt.c b/libparted/labels/gpt.c
index 05847de..ffb9fa0 100644
--- a/libparted/labels/gpt.c
+++ b/libparted/labels/gpt.c
@@ -922,11 +922,13 @@ gpt_read_headers (PedDisk const *disk,
 
   bool valid_primary = _header_is_valid (disk, pri, 1);
   if (valid_primary)
-    *primary_gpt = pri;
+    {
+      *primary_gpt = pri;
+      gpt_disk_data->AlternateLBA = PED_LE64_TO_CPU (pri->AlternateLBA);
+    }
   else
     pth_free (pri);
 
-  gpt_disk_data->AlternateLBA = PED_LE64_TO_CPU (pri->AlternateLBA);
   if( !valid_primary || gpt_disk_data->AlternateLBA > dev->length - 1 )
     gpt_disk_data->AlternateLBA = dev->length - 1;
 
-- 
2.1.0

